redisinsight/ui/src/components/base/utils/resize-observer/ResizeObserver.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useRef } from 'react'
+
+interface ResizeObserverProps {
+  children: (ref: React.Ref<HTMLDivElement>) => React.ReactNode
+  onResize: (dimensions: { height: number; width: number }) => void
+}
+
+export const ResizeObserver: React.FC<ResizeObserverProps> = ({
+  children,
+  onResize,
+}) => {
+  const containerRef = useRef<HTMLDivElement | null>(null)
+
+  useEffect(() => {
+    const element = containerRef.current
+    if (!element) return
+
+    const observer = new window.ResizeObserver(([entry]) => {
+      const { width, height } = entry.contentRect
+      onResize({ width, height })
+    })
+
+    observer.observe(element)
+
+    // eslint-disable-next-line consistent-return
```

Comment:
You can avoid esllint disable by reversing the conditional, i like early returns as well, but in this case I might go the other way
```suggestion
    if (element) {

    const observer = new window.ResizeObserver(([entry]) => {
      const { width, height } = entry.contentRect
      onResize({ width, height })
    })

    observer.observe(element)
}
```

Reply:
makes sense. no I'm getting the "not all paths return a value" warning, but I guess it is better.